api/pacman-api-admin/src/main/java/com/tmobile/pacman/api/admin/repository/service/RedHatAccountServiceImpl.java

```diff
@@ -0,0 +1,239 @@
+package com.tmobile.pacman.api.admin.repository.service;
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicSessionCredentials;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.secretsmanager.AWSSecretsManager;
+import com.amazonaws.services.secretsmanager.AWSSecretsManagerClientBuilder;
+import com.amazonaws.services.secretsmanager.model.CreateSecretRequest;
+import com.amazonaws.services.secretsmanager.model.CreateSecretResult;
+import com.amazonaws.services.secretsmanager.model.DeleteSecretRequest;
+import com.amazonaws.services.secretsmanager.model.DeleteSecretResult;
+import com.tmobile.pacman.api.admin.domain.AccountValidationResponse;
+import com.tmobile.pacman.api.admin.domain.CreateAccountRequest;
+import com.tmobile.pacman.api.admin.repository.model.AccountDetails;
+import com.tmobile.pacman.api.commons.Constants;
+import com.tmobile.pacman.api.commons.config.CredentialProvider;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.HttpClients;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpHeaders;
+import org.springframework.stereotype.Service;
+
+import java.net.UnknownHostException;
+import java.util.List;
+
+@Service
+public class RedHatAccountServiceImpl extends AbstractAccountServiceImpl implements AccountsService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RedHatAccountServiceImpl.class);
+
+    @Value("${secret.manager.path}")
+    private String secretManagerPrefix;
+    @Autowired
+    CredentialProvider credentialProvider;
+    private static final String ERROR_MESSAGE_TEMPLATE = "Error in %s account. ";
```

Comment:
I am sure that most of these are reusable across all Service Implementations, can all common ones be in Abstract class? this would avoid duplication.

Reply:
Yes, but currently its only used in redhat so I didnt moved it